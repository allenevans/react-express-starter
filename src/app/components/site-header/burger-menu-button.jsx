import PropTypes from 'prop-types';
import React, { Component, Fragment } from 'react';

import './burger-menu-button.scss';

class BurgerMenuButton extends Component {
  constructor(props) {
    super(props);

    this.state = {
      open: false,
    };
  }

  componentWillUnmount() {
    document.removeEventListener('click', this.handleClick, false);
  }

  handleClick = (event) => {
    event.preventDefault();
    this.toggleMenu(event, false);
    document.removeEventListener('click', this.handleClick, false);
  };

  listenForClick() {
    document.addEventListener('click', this.handleClick, false);
  }

  toggleMenu(event, open) {
    event.preventDefault();
    this.setState({ open }, () => this.listenForClick());
  }

  render() {
    return <Fragment>
      <label
        htmlFor="burger-menu-button__toggle" className="burger-menu-button__label"
        ref={(node) => { this.node = node; }}
        onClick={event => this.toggleMenu(event, !this.state.open)} />
      <input type="checkbox" id="burger-menu-button__toggle" className="burger-menu-button__toggle" checked={this.state.open} />
    </Fragment>;
  }
}

BurgerMenuButton.propTypes = {
  onOpenChange: PropTypes.func,
};

export default BurgerMenuButton;
