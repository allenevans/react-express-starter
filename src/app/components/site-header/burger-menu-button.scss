@import '../../../client/colors';

.burger-menu-button {
  &__label {
    background: linear-gradient(to bottom, $black-color 0%, $black-color 10%, transparent 10%, transparent 45%, $black-color 45%, $black-color 60%, transparent 60%, transparent 90%, $black-color 90%, $black-color 100%); //Gradient Hamburger
    border: 1px solid transparent;
    cursor: pointer;
    max-height: 1rem;
    max-width: 1.5rem;

    &:before {
      content: ' ';
      padding: 1rem;
    }
  }

  &__toggle {
    display: none;
  }
}
