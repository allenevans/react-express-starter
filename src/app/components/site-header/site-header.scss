@import '../../../client/breakpoints';
@import '../../../client/colors';
@import '../../../client/variables';

.site-header {
  $MOBILE_HEIGHT: 3rem;
  $TABLET_HEIGHT: 4rem;
  $PEEK_HEIGHT: 1rem;

  display: flex;
  flex: 1 0 100%;
  height: $MOBILE_HEIGHT;
  position: sticky;
  top: 0;
  align-items: center;
  flex-wrap: wrap;
  transition: top 0.5s;

  background: $white-color;

  &:before, &:after {
    content: ' ';
    width: $GUTTER_WIDTH_MOBILE;

    @include for-tablet-portrait-up {
      width: $GUTTER_WIDTH_TABLET;
    }
  }

  & > * {
    flex: 1;
  }

  nav {
    display: flex;

    & > * {
      flex: 1;
    }
  }

  &--peek {
    top: -$MOBILE_HEIGHT + $PEEK_HEIGHT;
  }

  @include for-tablet-portrait-up {
    height: $TABLET_HEIGHT;

    &--peek {
      top: -$TABLET_HEIGHT + $PEEK_HEIGHT;
    }
  }

  &__header-logo {
    color: $text-color;
    display: flex;
    align-items: center;
    font-family: 'Philosopher';
    font-size: $MOBILE_HEIGHT * .66;
    line-height: $MOBILE_HEIGHT;
    text-decoration: none;
    &--italic {
      font-style: italic;
    }

    @include for-tablet-portrait-up {
      font-size: $TABLET_HEIGHT * .66;
      line-height: $TABLET_HEIGHT;
    }
  }

  &__image-logo {
    flex: 0;
    max-height: $MOBILE_HEIGHT / 1.5;

    @include for-tablet-portrait-up {
      max-height: $TABLET_HEIGHT / 1.5;
    }
  }

  @include for-phone-only {
    .site-header__menu-nav {
      flex-wrap: wrap;
      position: absolute;

      & > a {
        flex: 1 0 100%;
        padding: 1rem;
        box-sizing: border-box;
        text-align: center;
        width: 100%;
      }
    }

    #burger-menu-button__toggle ~ .site-header__menu-nav {
      visibility: hidden;
    }

    #burger-menu-button__toggle:checked ~ .site-header__menu-nav {
      visibility: visible;
      height: auto;
      background-color: transparentize($white-color, 0.1);
      right: 0;
      top: $MOBILE_HEIGHT;
      width: 100%;
    }
  }

  @include for-tablet-portrait-up {
    .burger-menu-button__label, .burger-menu-button__toggle {
      display: none;
    }

    .site-header__menu-nav {
      flex: 1;
      height: 100%;
      align-items: center;

      & > a {
        height: 100%;
        text-align: center;
        line-height: $TABLET_HEIGHT;
      }
    }
  }
}
