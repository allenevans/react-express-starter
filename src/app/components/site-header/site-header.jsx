/* eslint-disable class-methods-use-this */
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import './site-header.scss';
import BurgerMenuButton from './burger-menu-button';
import logo from './logo.png';

export class SiteHeader extends Component {
  render() {
    return (
      <header className="site-header">
        <nav>
          <Link to="/" className="site-header__header-logo">
            <img src={logo} alt="Site logo" className="site-header__image-logo"/>
            Header
          </Link>
        </nav>

        <BurgerMenuButton />
        <nav className="site-header__menu-nav">
          <Link to="/about">About</Link>
        </nav>
      </header>
    );
  }
}

const mapStateToProps = () => ({
  actions: null,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({ }, dispatch),
});

SiteHeader.propTypes = {
  actions: PropTypes.object.isRequired,
};

export default connect(mapStateToProps, mapDispatchToProps)(SiteHeader);
