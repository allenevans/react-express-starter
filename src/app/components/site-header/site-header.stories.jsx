import React from 'react';
import { storiesOf } from '@storybook/react';
import { MemoryRouter } from 'react-router-dom';
import { Viewport, withViewport } from '@storybook/addon-viewport/preview';
import { SiteHeader } from './site-header';
import devices from '../../../../config/supported-devices';

const siteHeaderStories = storiesOf('Site Header', module);

devices.forEach(device => siteHeaderStories
  .addDecorator(withViewport(device))
  .addDecorator(story => <MemoryRouter>
    {story()}
  </MemoryRouter>)
  .add(`${device}`, () => {
    const containerStyle = {
      minHeight: '200vh',
      display: 'block',
    };

    return <Viewport>
      <style type="text/css">{`
          body {
            background-color: #ccc;
        `}</style>

      <div style={containerStyle} id="container">
        <SiteHeader actions={{}}/>
      </div>
    </Viewport>;
  }));
