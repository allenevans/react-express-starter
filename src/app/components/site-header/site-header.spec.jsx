import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import { SiteHeader } from './site-header';


describe('SiteHeader component', () => {
  const renderer = new ReactShallowRenderer();

  it('should render the component', () => {
    const actions = {};

    const tree = renderer.render(<SiteHeader actions={actions}/>);

    expect(tree).toMatchSnapshot();
  });
});
