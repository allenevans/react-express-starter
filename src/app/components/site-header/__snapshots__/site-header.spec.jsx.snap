// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteHeader component should render the component 1`] = `
<header
  className="site-header"
>
  <nav>
    <Link
      className="site-header__header-logo"
      replace={false}
      to="/"
    >
      <img
        alt="Site logo"
        className="site-header__image-logo"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACEAAAAeCAYAAACiyHcXAAAABGdBTUEAALGPC/xhBQAAACBjSFJNAAB6JgAAgIQAAPoAAACA6AAAdTAAAOpgAAA6mAAAF3CculE8AAAACXBIWXMAAAsTAAALEwEAmpwYAAACC2lUWHRYTUw6Y29tLmFkb2JlLnhtcAAAAAAAPHg6eG1wbWV0YSB4bWxuczp4PSJhZG9iZTpuczptZXRhLyIgeDp4bXB0az0iWE1QIENvcmUgNS40LjAiPgogICA8cmRmOlJERiB4bWxuczpyZGY9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkvMDIvMjItcmRmLXN5bnRheC1ucyMiPgogICAgICA8cmRmOkRlc2NyaXB0aW9uIHJkZjphYm91dD0iIgogICAgICAgICAgICB4bWxuczp0aWZmPSJodHRwOi8vbnMuYWRvYmUuY29tL3RpZmYvMS4wLyI+CiAgICAgICAgIDx0aWZmOlJlc29sdXRpb25Vbml0PjI8L3RpZmY6UmVzb2x1dGlvblVuaXQ+CiAgICAgICAgIDx0aWZmOkNvbXByZXNzaW9uPjE8L3RpZmY6Q29tcHJlc3Npb24+CiAgICAgICAgIDx0aWZmOk9yaWVudGF0aW9uPjE8L3RpZmY6T3JpZW50YXRpb24+CiAgICAgICAgIDx0aWZmOlBob3RvbWV0cmljSW50ZXJwcmV0YXRpb24+MjwvdGlmZjpQaG90b21ldHJpY0ludGVycHJldGF0aW9uPgogICAgICA8L3JkZjpEZXNjcmlwdGlvbj4KICAgPC9yZGY6UkRGPgo8L3g6eG1wbWV0YT4KD0UqkwAABUlJREFUSA21Vl1sFFUU/u6dme1O291ud+kPqUhIWgnRRsRALRIVeQGC0FRaTYgKJhhFikZ+4gOE5a2++aCJz/hCtq8GE2MioAaICbz4F0IiAYISgqS4tOzOzozfme42Y3e7OxviSWb3zrnn57vnnu/eAR5BfEBV3MPjii7qvxnVcKGdJCUCP5dEPzxMqzzuVHQLbf+X9yygJfC7Nvp+68GVX7pxUt4fpRri35QwWQDi8hJ8cr0X/pVeFM5lsEOCVOaaCRgEa8YhBxjcBu/bDNa2GXgn78PnnsbSBo4wTlLmmq1IsyDUOOAymV5i4OOYQopj76EPz9ZYfz6D3XzHVLlSMo4iTYFgFQL7r9J4Na4wMsMlUwKdDDMmJk52YIUAzTYBJDIIKbEEfz+BzAoLhy0VJPeYW2iqiz7cFoX+VTEc4DuOy09EiQyC8QyJuSuOvbbC2gK3gK9hf81tQUJh96k0niv3Rnhe3GtKJKMskzFo6esEVqY09pWk9tWi2CxuXCP1dAyHOC2gxVIqVVcigWBpg7SP2/igTWGZw2Zk1CpfGhmzwhYfI993YZTZfeoeHUSZkv7pDF5mOd6UZmTUmoFFyWmfHDY6dVCNTuoaUrZqNQvqNkfJsbFYr4FD7IU2JhGK1gRR9pXe8FoV1l3sxp6yrp59dUnDIHJjYwHIS1NTr7eb1uYZDz6jNQIehBCkKYWJL3m30MfL1vFbFKHv+4riY9/u3h9/v/rN0vM/DD7stDy4TiQQxOCyf4y7Pj5bcxsTsotMxr9qWTTgiRMnAoCT/asnrm7eNjjd95irZh3t64Cp1ZGqNZpNilbgrVwnNggAvtbMV1OZzWY1H4/P6oRl7P07mcKNTVs07nE5pl2drrZGkUIuT9bEIA83mvCUD1gVLC7sUguEEgBilEgkDpuW1aUdx73+xEp1f3gd9O08ORgLx6g3NsgmYcsrZ5dgpxhWjv6wUxWIsXIzHj16dLtlWTsLxSIM39dFuxXXnt8Aryiba4VjNBr7poJKaxykYUaOfumPsNNCEGpqakoau72np+dgjMIG9VgWZZZK+HPZctzZtgX65gP4VuRtCSjbrrHmQhf2lpMvDiKXywWgJicn9xiG8UKxWPTIkECn2Fa+aeLas2vhJAzoEs/NuanwomqOZeUOf3iA7T+VwSrpjWyoSecrwT7Q4+Pj7tatW5fH4/HgJgxHJGEh1bjb1Y1b28eh/yqxGuz9CMKkqjTXpH0DBj4Ul+Mhus6DoD746N24ceN+gugvlUqyLeF58Q0U154aRP7JAej7ed4Wkb+VtRz5vGV3XU7hJQKbp2yQhFUw+RSPHTs2ZNv22wRQySf/8yLVMFwX+WQHbry4CZjmlBGfn28w4MkHlzxta4nhCG3J3jnKhldqplKpw2RE2vO8Re+HYFvYqjf7BzC9YRj6llC2pUH++emAsvwg2nI2g9dES2BaDiWpQkkoSTLsKBQKpGD9jtMEUWyx8Qcpy8KQb7IWriua+PJx0mXio0+70cOha5w5c8YbGhpKDg8Pf84qLK9XhUoOguTZ4eGfjg4kYwaSP/0KP80mdcmYxiIfP3LLLu31cf+LBzgXbMfIyMh7BLDecRypgsGnrogRV0AkpiJlVbEvrYzCjNJaSwkbPvQ1BC6b4sDpLjxjjI6OLh0YGJhk2AyfGc5JV4qNw/f/PGEdgTjcFudBW5vTri2n48LPjtdpO8rlSsr+i/0ThMMDsMir0E4D35nJZPLe7OzsG62trabrujwePdpEEzm8PctC3o5B0AefO8GtU9+ffj5vWG3ZncVLRsfVfwGAHa53vGvQ8wAAAABJRU5ErkJggg=="
      />
      Header
    </Link>
  </nav>
  <BurgerMenuButton />
  <nav
    className="site-header__menu-nav"
  >
    <Link
      replace={false}
      to="/about"
    >
      About
    </Link>
  </nav>
</header>
`;
